nukebuild/Build.cs

```diff
@@ -407,6 +408,67 @@ await Task.WhenAll(
             file.GenerateCppHeader());
     });
 
+    Target VerifyXamlCompilation => _ => _
+        .DependsOn(CreateNugetPackages)
+        .Executes(() =>
+        {
+            var buildTestsDirectory = RootDirectory / "tests" / "BuildTests";
+            var artifactsDirectory = buildTestsDirectory / "artifacts";
+            var nugetCacheDirectory = artifactsDirectory / "nuget-cache";
+
+            DeleteDirectory(artifactsDirectory);
+            BuildTestsAndVerify("Debug");
+            BuildTestsAndVerify("Release");
+
+            void BuildTestsAndVerify(string configuration)
+            {
+                var configName = configuration.ToLowerInvariant();
+
+                DotNetBuild(settings => settings
+                    .SetConfiguration(configuration)
+                    .SetProperty("AvaloniaVersion", Parameters.Version)
+                    .SetProperty("NuGetPackageRoot", nugetCacheDirectory)
+                    .SetPackageDirectory(nugetCacheDirectory)
+                    .SetProjectFile(buildTestsDirectory / "BuildTests.sln")
+                    .SetProcessArgumentConfigurator(arguments => arguments.Add("--nodeReuse:false")));
+
+                // Standard compilation - should have compiled XAML
+                VerifyBuildTestAssembly("bin", "BuildTests");
+                VerifyBuildTestAssembly("bin", "BuildTests.Android");
```

Comment:
Note that the original issue can't be reproduced with a plain `dotnet build`, with Rider it was only happened when one tries to debug the app.